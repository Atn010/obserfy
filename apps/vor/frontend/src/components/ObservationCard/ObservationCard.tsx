import React, { FC } from "react"
import Flex from "../Flex/Flex"
import Typography from "../Typography/Typography"
import Pill from "../Pill/Pill"
import Spacer from "../Spacer/Spacer"
import Button from "../Button/Button"
import Card from "../Card/Card"
import { categories } from "../../categories"
import { Observation } from "../../api/useGetObservations"

interface Props {
  observation: Observation
  onDelete: (value: Observation) => void
  onEdit: (value: Observation) => void
}
export const ObservationCard: FC<Props> = ({
  onDelete,
  onEdit,
  observation,
}) => {
  const category = categories[parseInt(observation.categoryId, 10)]

  return (
    <Card mb={2} key={observation.id} borderRadius={[0, "default"]}>
      <Typography.H6
        mx={3}
        mt={observation.longDesc ? 3 : 2}
        mb={0}
        data-cy="observation-short-desc"
      >
        {observation.shortDesc}
      </Typography.H6>
      {observation.longDesc && (
        <Typography.Body
          fontSize={1}
          mt={2}
          mx={3}
          mb={3}
          data-cy="observation-long-desc"
        >
          {observation.longDesc}
        </Typography.Body>
      )}
      <Flex
        p={2}
        pb={0}
        alignItems="center"
        sx={{
          borderTopWidth: 1,
          borderTopStyle: observation.longDesc ? "solid" : "none",
          borderTopColor: "border",
        }}
      >
        <Flex flexWrap="wrap">
          <Pill
            ml={2}
            mb={2}
            backgroundColor={category.color}
            text={category.name}
            color={category.onColor}
          />
          {observation.creatorName && (
            <Pill
              ml={2}
              mb={2}
              text={observation.creatorName.split(" ")[0]}
              color="text"
            />
          )}
        </Flex>
        <Spacer />
        <Button
          variant="secondary"
          color="danger"
          data-cy="delete-observation"
          onClick={() => onDelete(observation)}
          fontSize={0}
          mb={2}
        >
          delete
        </Button>
        <Button
          variant="secondary"
          data-cy="edit-observation"
          fontSize={0}
          onClick={() => onEdit(observation)}
          mb={2}
        >
          Edit
        </Button>
      </Flex>
    </Card>
  )
}

export default ObservationCard
