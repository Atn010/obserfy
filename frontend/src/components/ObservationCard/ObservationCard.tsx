import React, { FC } from "react"
import differenceInCalendarDays from "date-fns/differenceInCalendarDays"
import Flex from "../Flex/Flex"
import Typography from "../Typography/Typography"
import Pill from "../Pill/Pill"
import Spacer from "../Spacer/Spacer"
import Button from "../Button/Button"
import Card from "../Card/Card"
import { categories } from "../../categories"
import { Observation } from "../../hooks/students/useQueryStudentObservations"
import { useFormatRelativeTime } from "../../intl/useFormatDate"

interface Props {
  observation: Observation
  onDelete: (value: Observation) => void
  onEdit: (value: Observation) => void
}
export const ObservationCard: FC<Props> = ({
  onDelete,
  onEdit,
  observation,
}) => {
  const category = categories[parseInt(observation.categoryId, 10)]

  const createdDateDifference = differenceInCalendarDays(
    Date.parse(observation.createdDate ?? ""),
    Date.now()
  )

  const relativeCreatedDate = useFormatRelativeTime(
    createdDateDifference,
    "day",
    { numeric: "auto" }
  )

  return (
    <Card mb={3} key={observation.id}>
      <Flex
        p={3}
        alignItems="center"
        sx={{
          borderBottomWidth: 1,
          borderBottomColor: "border",
          borderBottomStyle: "solid",
        }}
      >
        <Flex flexDirection="column" alignItems="start">
          <Typography.H6 mb={3}>{observation.shortDesc}</Typography.H6>
          <Flex>
            <Pill
              backgroundColor={category.color}
              text={category.name}
              color={category.onColor}
            />
            <Pill ml={2} text={relativeCreatedDate} color="text" />
          </Flex>
        </Flex>
      </Flex>
      {observation.longDesc && (
        <Typography.Body fontSize={1} p={3}>
          {observation.longDesc}
        </Typography.Body>
      )}
      <Flex
        p={2}
        alignItems="center"
        sx={{
          borderTopWidth: 1,
          borderTopStyle: observation.longDesc ? "solid" : "none",
          borderTopColor: "border",
        }}
      >
        <Spacer />
        <Button
          variant="secondary"
          color="danger"
          onClick={() => onDelete(observation)}
          fontSize={0}
        >
          delete
        </Button>
        <Button
          variant="secondary"
          data-cy="dialogPositiveAction"
          onClick={() => onEdit(observation)}
          fontSize={0}
        >
          Edit
        </Button>
      </Flex>
    </Card>
  )
}

export default ObservationCard
