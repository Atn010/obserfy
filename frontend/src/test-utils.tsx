/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { IntlProvider } from "gatsby-plugin-intl3"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"
import en from "./intl/en.json"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <IntlProvider locale="en" messages={en}>
          <Box fontSize={[16, 20]}>{children}</Box>
        </IntlProvider>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
