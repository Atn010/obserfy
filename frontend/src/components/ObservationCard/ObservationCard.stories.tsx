import React, { FC } from "react"
import { action } from "@storybook/addon-actions"
import { text } from "@storybook/addon-knobs"
import ObservationCard from "./ObservationCard"

export default {
  title: "Core|ObservationCard",
  component: ObservationCard,
  parameters: {
    componentSubtitle: "Just a simple ObservationCard",
  },
}

export const Basic: FC = () => (
  <ObservationCard
    observation={{
      studentName: "Angelica",
      createdDate: text("date", "2019-12-10T11:53:02.050339Z"),
      id: "",
      studentId: "",
      longDesc: "",
      categoryId: "0",
      shortDesc: "",
    }}
    onDelete={action("delete")}
    onEdit={action("edit")}
  />
)
