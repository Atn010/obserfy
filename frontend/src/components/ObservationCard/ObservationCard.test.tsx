import React from "react"
import { render } from "../../test-utils"
import { Basic } from "./ObservationCard.stories"

describe("ObservationCard", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
