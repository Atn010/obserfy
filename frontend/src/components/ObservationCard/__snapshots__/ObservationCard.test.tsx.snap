// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationCard should render correctly 1`] = `
.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width:40em) {
  .emotion-14 {
    font-size: 20px;
  }
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 8px;
  background-color: var(--theme-ui-colors-background,#F7F8FA);
  border-radius: 6px;
  box-shadow: rgba(0,0,0,0.1) 0px 0px 1px,rgba(0,0,0,0.27) 0px 2px 4px -2px;
  padding: 0;
  background-color: var(--theme-ui-colors-surface,#fff);
  overflow: hidden;
  margin-bottom: 8px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-bottom-width: 1px;
  border-bottom-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  border-bottom-style: solid;
  padding: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 24px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 300;
  line-height: heading;
  font-size: 2rem;
  line-height: 3rem;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  margin-bottom: 8px;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 99999px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  background-color: #b39ddb;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  text-transform: capitalize;
  color: rgba(0,0,0,0.87);
  font-size: 12px;
  line-height: 1em;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 99999px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  margin-left: 8px;
  background-color: #b39ddb;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  font-size: 14px;
  padding: 16px;
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-top-width: 1px;
  border-top-style: solid;
  border-top-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  padding: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  pointer-events: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary,#6E55C6);
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  text-transform: capitalize;
  box-sizing: border-box;
  font-size: 14px;
  font-weight: bold;
  color: var(--theme-ui-colors-textPrimary,#3A2B95);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  border: 1px solid;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 16px;
  color: var(--theme-ui-colors-danger,#d50000);
}

.emotion-10:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-10:hover,
.emotion-10:focus {
  background-color: var(--theme-ui-colors-primaryLighter,rgba(161,130,250,0.12));
  border-color: var(--theme-ui-colors-primary,#6E55C6);
  outline: none;
}

.emotion-10:hover:disabled,
.emotion-10:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary,#6E55C6);
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  text-transform: capitalize;
  box-sizing: border-box;
  font-size: 14px;
  font-weight: bold;
  color: var(--theme-ui-colors-textPrimary,#3A2B95);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  border: 1px solid;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-11:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-11:hover,
.emotion-11:focus {
  background-color: var(--theme-ui-colors-primaryLighter,rgba(161,130,250,0.12));
  border-color: var(--theme-ui-colors-primary,#6E55C6);
  outline: none;
}

.emotion-11:hover:disabled,
.emotion-11:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
}

<div>
  <div
    class="emotion-14"
  >
    <div
      class="emotion-13"
    >
      <div
        class="emotion-7"
      >
        <div
          class="emotion-6"
        >
          <h6
            class="emotion-0"
          />
          <div
            class="emotion-5"
          >
            <div
              class="emotion-2"
            >
              <p
                class="emotion-1"
              >
                Others
              </p>
            </div>
            <div
              class="emotion-4"
            >
              <p
                class="emotion-1"
              >
                yesterday
              </p>
            </div>
          </div>
        </div>
      </div>
      <p
        class="emotion-8"
      />
      <div
        class="emotion-12"
      >
        <div
          class="emotion-9"
        />
        <button
          class="emotion-10"
        >
          delete
        </button>
        <button
          class="emotion-11"
          data-cy="dialogPositiveAction"
        >
          Edit
        </button>
      </div>
    </div>
  </div>
</div>
`;
